File: 1013-partition-array-into-three-parts-with-equal-sum/1013-partition-array-into-three-parts-with-equal-sum.java
class Solution {
    public boolean canThreePartsEqualSum(int[] arr) {
        int n=arr.length;
        int sum=0,firstSum=0,part=0;
        for(int i=0;i<n;i++){
            sum+=arr[i];
        }
        firstSum=sum/3;
        int j=0,c=0;
        while(j<n){
            part+=arr[j];
            if(part==firstSum){
                c++;
                part=0;
            }
            j++;
        }
        return c>=3 && sum%3==0;
    }
}

File: 1013-partition-array-into-three-parts-with-equal-sum/1013-partition-array-into-three-parts-with-equal-sum.py
class Solution:
    def canThreePartsEqualSum(self, arr: List[int]) -> bool:
        sum,part=0,0
        for num in arr:
            sum+=num
        firstSum=sum//3
        c=0
        for num in arr:
            part+=num
            if(part==firstSum):
                part=0
                c+=1
        return c>=3 and sum%3==0

        